### PR TITLE
feat(api): Add automatically connect option

### PR DESCRIPTION
**Pull request by jczstudios**
If set to false, the API server will not automatically call connect. This can used to change options after construction of the client or only connect on some instances when sharded.

### CODE REVIEW

File: packages/api/src/register.ts [hunk 1]
```diff
@@ -28,6 +28,10 @@ export class Api extends Plugin {
 	 * @since 1.0.0
 	 */
 	public static async [preLogin](this: SapphireClient): Promise<void> {
+		if (this.server.options.automaticallyConnect === false) {
+			return Promise.resolve();
+		}
```
Comment (favna) [inline packages/api/src/register.ts:33]:
```suggestion
		if (!(server.options.automaticallyConnect ?? true)) {
			return;
		}
```

File: packages/api/src/lib/structures/http/Server.ts [hunk 2]
```diff
@@ -219,6 +219,13 @@ export interface ServerOptions {
 	 * @default {}
 	 */
 	auth?: ServerOptionsAuth;
+
+	/**
+	 * Should the server automatically connect when the plugin is loaded.
```
Comment (favna) [inline packages/api/src/lib/structures/http/Server.ts:224]:
```suggestion
	 * Whether the server should connect upon being when the plugin is loaded.
```

File: packages/api/src/lib/structures/http/Server.ts [hunk 3]
```diff
@@ -219,6 +219,13 @@ export interface ServerOptions {
...
+	 * @since 3.1.4
```
Comment (favna) [inline packages/api/src/lib/structures/http/Server.ts:225]:
```suggestion
	 * @since 3.2.0
```

**Review by favna — CHANGES_REQUESTED** [review]
This will work better

https://user-images.githubusercontent.com/4019718/156416939-2b1da1cf-5a97-48a5-8954-1a3161e74b4a.mp4

**Comment by jczstudios** [comment]
Yeah, that's better. Thanks.

**Review by favna — APPROVED** [review]

**Comment by favna** [comment]
@sapphiredev pack

**Review by kyranet — APPROVED** [review]

**Pull request merged**